Feature/load stream queue

TwilCynder opened the pull request:
Making TSH able to load the current Stream Queue into the program state. 

Currently : 
- Loads when a set is selected

Possible improvements
- Manual update button
- UI to edit the stream queue

Wolfy76700 commented:
So I’ve tested it using [this test bracket](https://www.start.gg/tournament/tsh-test-tournament-1/event/incomplete-top-12/overview) and couldn’t load the queue properly. Instead I got this error message in the console:

```
==================================GetStreamQueue=====================================
=======================================UpdateStreamQueue==================================
{}
{}
Traceback (most recent call last):
  File "C:\Users\wolfy\Git\TournamentStreamHelper\src\TournamentDataProvider\StartGGDataProvider.py", line 774, in ParseMatchDataOldApi
    for st in allStages:
TypeError: 'NoneType' object is not iterable

Get recent sets start
Recent sets size: 0
{}
Traceback (most recent call last):
  File "C:\Users\wolfy\Git\TournamentStreamHelper\src\TournamentDataProvider\StartGGDataProvider.py", line 774, in ParseMatchDataOldApi
    for st in allStages:
TypeError: 'NoneType' object is not iterable

{}
Traceback (most recent call last):
  File "C:\Users\wolfy\Git\TournamentStreamHelper\src\TournamentDataProvider\StartGGDataProvider.py", line 774, in ParseMatchDataOldApi
    for st in allStages:
TypeError: 'NoneType' object is not iterable

{}
Traceback (most recent call last):
  File "C:\Users\wolfy\Git\TournamentStreamHelper\src\TournamentDataProvider\StartGGDataProvider.py", line 774, in ParseMatchDataOldApi
    for st in allStages:
TypeError: 'NoneType' object is not iterable

```

I get similar results both when TSH has the stream channel indicated and when it doesn’t

Wolfy76700 commented:
OK after further testing I was able to load both stream queues with a bit of fiddling, although in a weird order due to how the queue itself was created

Side note, could we add a small UI element similar to the sets UI indicating the update rate and a start / pause button? This would probably help with users forgetting to set their stream before selecting a set

(Also @joaorb64: If we select a stream in a given queue, should the program automatically set the stream channel to the one from the queue?)

![Screenshot_1303](https://github.com/joaorb64/TournamentStreamHelper/assets/1964201/2ff28156-b30d-4867-a7d9-c41fa50e2989)

ZombieHDGaming commented:
I'll piggyback off Mathias' comment with the data structure. I think I have an easier way to separate the data, while still maintaining compatibility, and making it easy for people. I believe the data could look something like:

> Data.stream_queue.[STREAM NAME].<NUMBER>

With the above structure, it should allow us to easily sort the data for events with multiple streams and queues, but also makes it easy to pull in data for the currently set stream name. Mathias might be right and the data will need to be adjusted to more fall in line with the scoreboard player info. As for names not being localized, we call always fix that before the merge.

TwilCynder commented:
> So I’ve tested it using [this test bracket](https://www.start.gg/tournament/tsh-test-tournament-1/event/incomplete-top-12/overview) and couldn’t load the queue properly. Instead I got this error message in the console:
> 
> ```
> ==================================GetStreamQueue=====================================
> =======================================UpdateStreamQueue==================================
> {}
> {}
> Traceback (most recent call last):
>   File "C:\Users\wolfy\Git\TournamentStreamHelper\src\TournamentDataProvider\StartGGDataProvider.py", line 774, in ParseMatchDataOldApi
>     for st in allStages:
> TypeError: 'NoneType' object is not iterable
> 
> Get recent sets start
> Recent sets size: 0
> {}
> Traceback (most recent call last):
>   File "C:\Users\wolfy\Git\TournamentStreamHelper\src\TournamentDataProvider\StartGGDataProvider.py", line 774, in ParseMatchDataOldApi
>     for st in allStages:
> TypeError: 'NoneType' object is not iterable
> 
> {}
> Traceback (most recent call last):
>   File "C:\Users\wolfy\Git\TournamentStreamHelper\src\TournamentDataProvider\StartGGDataProvider.py", line 774, in ParseMatchDataOldApi
>     for st in allStages:
> TypeError: 'NoneType' object is not iterable
> 
> {}
> Traceback (most recent call last):
>   File "C:\Users\wolfy\Git\TournamentStreamHelper\src\TournamentDataProvider\StartGGDataProvider.py", line 774, in ParseMatchDataOldApi
>     for st in allStages:
> TypeError: 'NoneType' object is not iterable
> ```
> 
> I get similar results both when TSH has the stream channel indicated and when it doesn’t

These error messages were already here, i don't really know why they're here but the function that raises these exceptions have no connection with my code

I'll try to fix the data structure issue and then move on to the UI

Wolfy76700 commented:
Yeah this is related to stages apparently, we might need further investigation on that at some point

TwilCynder commented:
Just changed the data structure in the program state to include more info + match the structure of data.score.  
The stream queue for every stream is included ; i also added data.currentStream so a "current stream queue" layout can know which stream queue to display.
